src/Uno.UI/DataBinding/BindingPropertyHelper.cs

```diff
@@ -110,9 +110,9 @@ public static bool IsEvent(Type type, string property)
 
 		public static Type? GetPropertyType(Type type, string property, bool allowPrivateMembers)
 		{
-			var key = CachedTuple.Create(type, property, allowPrivateMembers);
+			var key = new GetPropertyTypeKey(type, property, allowPrivateMembers);
```

Comment:
Is this code path supposed to be executed only on the UI thread?
If so, I think we could change `GetPropertyTypeKey` to be mutable and have a static key, then we just mutate the existing key and use that for getting the value from the dictionary. This will make `GetPropertyType` not allocate a new `GetPropertyTypeKey` each time.

And also, if it's expected to be on UI thread only, I don't see why we need to lock below.

Reply:
Yes, it's UI Thread only.

Reply:
I was worried about multiple threads accessing this one, let's do it !